`python39Packages.weasyprint` build failure on x86_64-linux as of `83b213ef`

## Issue description
Build of `python39Packages.weasyprint` failed on x86_64-linux as of 83b213efd12aedab721f02154123c54d3a1e4e7b. This is currently breaking `python3Packages.wandb`.

```
error: builder for '/nix/store/05qakhks185rql1731139av7abi9zkb9-python3.9-weasyprint-54.3.drv' failed with exit code 1;
       last 10 log lines:
       > INFO     weasyprint.progress:__init__.py:210 Step 2 - Fetching and parsing CSS - CSS string
       > INFO     weasyprint.progress:__init__.py:65 Step 3 - Applying CSS
       > INFO     weasyprint.progress:document.py:903 Step 4 - Creating formatting structure
       > INFO     weasyprint.progress:page.py:849 Step 5 - Creating layout - Page 1
       > INFO     weasyprint.progress:document.py:1132 Step 6 - Creating PDF
       > INFO     weasyprint.progress:document.py:1288 Step 7 - Adding PDF metadata
       > =========================== short test summary info ============================
       > FAILED tests/draw/test_gradient.py::test_linear_gradients_5 - AssertionError:...
       > FAILED tests/draw/test_gradient.py::test_linear_gradients_12 - AssertionError...
       > ===== 2 failed, 1594 passed, 6 deselected, 31 xfailed in 96.34s (0:01:36) ======
       For full logs, run 'nix log /nix/store/05qakhks185rql1731139av7abi9zkb9-python3.9-weasyprint-54.3.drv'.
error: 1 dependencies of derivation '/nix/store/ch0qvbz9kdn4m5wkh6rrr8sxfyl86dbp-python3.9-trytond-6.4.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/49sdq287sk4x7lcplrwwnzp2f0vaii7q-python3.9-sentry-sdk-1.5.11.drv' failed to build
error: 1 dependencies of derivation '/nix/store/9bvvljrarsyad6g0wwbp1j6swwlgl3ia-python3.9-wandb-0.12.16.drv' failed to build
```

[full build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2304050701)

python3Packages.wandb maintainers cc: @samuela
python39Packages.weasyprint maintainers cc: @elohmeier
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit 83b213efd12aedab721f02154123c54d3a1e4e7b
2. Run `nix-build -A python39Packages.weasyprint`

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.13.0-1022-azure, Ubuntu, 20.04.4 LTS (Focal Fossa), nobuild`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.8.0`
 - nixpkgs: `/nix/store/cds32qbc6r5c53zpai7q9gzg5fk8kwbf-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/2304050701) that created this issue.
- Internal tags: -8167462949332237166 8686099623583392054

--- a/pkgs/development/python-modules/weasyprint/default.nix
+++ b/pkgs/development/python-modules/weasyprint/default.nix
@@ -48,6 +48,12 @@ buildPythonPackage rec {
       pangocairo = "${pango.out}/lib/libpangocairo-1.0${stdenv.hostPlatform.extensions.sharedLibrary}";
       harfbuzz = "${harfbuzz.out}/lib/libharfbuzz${stdenv.hostPlatform.extensions.sharedLibrary}";
     })
+    # Disable tests for new Ghostscript
+    # Remove when next version is released
+    (fetchpatch {
+      url = "https://github.com/Kozea/WeasyPrint/commit/e544398b00d76bc0317ea7e2abe40dc46b380910.patch";
+      sha256 = "sha256-oQO3j9Mo1x98WaLPROxsOn0qkeYRJrCx5QWWKoHvabE=";
+    })
   ];
 
   nativeBuildInputs = [

